js/index.js

```diff
@@ -0,0 +1,925 @@
+'use strict';
+const MAX_LENGTH = 100;
+const eqTrue = /(.*)==( *)true(.*)/g;
+const eqFalse = /(.*)==( *)false(.*)/g;
+const scan = /(.*)new Scanner\(System\.in\)(.)*/g;
+const blankPrintln = /System\.out\.println\("[ ]*"\)/g;
+const state = {
+    lineErrors: [],
+    errorsByType: {}
+}
+
+let tabSize;
+let style;
+let inClass;
+
+window.onload = () => {     
```

Comment:
We normally suggest putting your JS after your HTMl, but this works fine (it's taught this way in CSE 154)